services/brig/src/Brig/API/Internal.hs

```diff
@@ -475,14 +474,20 @@ createUserNoVerifySpar uData =
 
 deleteUserNoVerifyH :: UserId -> (Handler r) Response
 deleteUserNoVerifyH uid = do
```

Comment:
What is the meaning of "no verify" in this case? Seems like we're doing some verification. Also should it make sense to retry internally a few times until we get "already deleted"?

Reply:
I haven't invented the name (only recycled :recycle: :wink:). However, to me it means that there is no verification that you're allowed to do so. I.e. when you call this with a valid `UserId`, the user will be deleted without any additional checks/constraints.

Reply:
As I've already got the code to check if a user has been deleted completely, a retry mechanism sounds quite reasonable. :thinking:

Reply:
> What is the meaning of "no verify" in this case? Seems like we're doing some verification.

you're right, it would be an improvement to rename this: `s/NoVerify/NoAuth/` and/or add a comment that this is the handler without the caller authentication/authorization part.

>  Also should it make sense to retry internally a few times until we get "already deleted"?

This handler queues the deletion for async procesing; IIRC (please double-check), the entry will be removed from the queue at the end of the deletion, so retry would merely trigger if the queue is somehow broken.  I don't think that's needed.

Reply:
@fisx I'll do the renaming. However, the processing isn't asynchronous anymore. I changed that so synchronous to get a result. (To be able to tell the SCIM user if their request failed.)